docs/topics/extensions.md

```diff
@@ -99,6 +99,68 @@ Or they can be referenced as a preprovision extension, this will run during clou
 |---|---|---|
 |name|yes|The name of the extension. This must match the name in the extensionProfiles|
 
+## agentPoolProfiles
+
+Extensions run a script on every node in an agent pool. The extensions array in the

```

Comment:
I believe there is a way to run extensions on only a single node.
See the `Extension.SingleOrAll` struct at https://github.com/Azure/aks-engine/blob/5dc9c4e67a8426e0249ac15c4ce8ccca2211d849/pkg/api/types.go#L561
It looks like this is used in https://github.com/Azure/aks-engine/blob/5dc9c4e67a8426e0249ac15c4ce8ccca2211d849/pkg/engine/vmextensions.go#L234 but I cannot confirm it works as expected.

Reply:
Let's merge this improvement for now and we can do test and document single node as a separate effort